lib/kubernetes-deploy/kubernetes_resource/custom_resource_definition.rb

```diff
@@ -41,6 +43,25 @@ def prunable?
       prunable == "true"
     end
 
+    def timeout_for_children
+      @definition.dig("metadata", "annotations", CHILD_CR_TIMEOUT_ANNOTATION)&.to_i
+    end
+
+    def rollout_params
+      return nil unless rollout_params_string
+
+      raw_params = JSON.parse(rollout_params_string)
+      params = {
+        success_queries: raw_params["success_queries"] || default_success_query,
+        failure_queries: raw_params["failure_queries"] || default_failure_query,
+      }.deep_symbolize_keys
+      # Preemptively check if the supplied JsonPath expressions are valid
+      params if validate_params(params)
+
+    rescue JSON::ParserError
+      raise FatalDeploymentError, "custom rollout params are not valid JSON: '#{rollout_params_string}'"
```

Comment:
Should we actually fail the whole deploy on this, or is there a more graceful fallback behaviour we could adopt?

Reply:
This is a value-judgement we'll need to make. My opinion is that, if users are opting-in to use this feature, we should consider it critical and fail fast if something goes wrong. 

On the other hand, aborting a deploy because of some bad JSON might cause too much friction.